toolchain/base/value_store.h

```diff
@@ -188,12 +258,10 @@ class SharedValueStores : public Printable<SharedValueStores> {
 
   auto Print(llvm::raw_ostream& out) const -> void {
     out << "shared_values:\n"
-        << "  - integers:\n";
-    integers_.Print(out, 6);
-    out << "  - reals:\n";
-    reals_.Print(out, 6);
-    out << "  - strings:\n";
-    strings_.Print(out, 6);
+        << "  - ";
+    integers_.Print(out, "integers", 0, 6);
+    reals_.Print(out, "reals", 4, 6);
+    strings_.Print(out, "strings", 4, 6);
```

Comment:
These magic numbers make me think that we could model this better -- maybe the entry should not be responsible for printing out its own first-line prefix, and maybe we could track the indent width automatically somehow, ...

Actually, maybe we should switch over to using the `llvm::yaml` machinery to do YAML printing?

Reply:
Separating out this change.